java-checks/src/main/java/org/sonar/java/symexecengine/DataFlowVisitor.java

```diff
@@ -43,25 +46,46 @@
 import org.sonar.plugins.java.api.tree.WhileStatementTree;
 
 import javax.annotation.CheckForNull;
+
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Set;
 
-public abstract class DataFlowVisitor extends BaseTreeVisitor {
+public class DataFlowVisitor extends BaseTreeVisitor {
+
+  private final SymbolicExecutionCheck check;
 
-  public DataFlowVisitor() {
-    executionState = new ExecutionState();
+  public static void analyze(MethodTree method, SymbolicExecutionCheck check) {
+    BlockTree block = method.block();
+    if (block != null) {
+      ExecutionState executionState = new ExecutionState();
+      List<SymbolicValue> arguments = new ArrayList<>();
+      for (VariableTree methodParameter : method.parameters()) {
+        executionState.defineSymbol(methodParameter.symbol());
+        arguments.add(executionState.createValueForSymbol(methodParameter.symbol(), methodParameter));
+      }
+      check.initialize(executionState, method, arguments);
+      block.accept(new DataFlowVisitor(executionState, method, check));
+      for (State state : executionState.getStatesOfCurrentExecutionState()) {
+        check.onValueUnreachable(executionState, state);
+      }
+    }
   }
 
-  protected ExecutionState executionState;
+  private DataFlowVisitor(ExecutionState executionState, MethodTree analyzedMethod, SymbolicExecutionCheck check) {
+    this.check = check;
+    this.executionState = executionState;
+  }
 
-  protected abstract boolean isSymbolRelevant(Symbol symbol);
+  private ExecutionState executionState;
 
   @Override
   public void visitVariable(VariableTree tree) {
     super.visitVariable(tree);
-    if (isSymbolRelevant(tree.symbol())) {
-      executionState.defineSymbol(tree.symbol());
-      executionState.createValueForSymbol(tree.symbol(), tree);
+    executionState.defineSymbol(tree.symbol());
+    executionState.createValueForSymbol(tree.symbol(), tree);
+    if (tree.initializer() != null) {
+      check.onAssignment(executionState, tree, tree.symbol(), tree.initializer());
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png) ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Method 'SymbolicExecutionCheck.onAssignment(...)' is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)